lib/IDE/Refactoring.cpp

```diff
@@ -2602,6 +2602,169 @@ bool RefactoringActionLocalizeString::performChange() {
+}
+
+void generateDocCommentForFunction(FuncDecl *funcDecl,
+                                   swift::SourceLoc startLocation,
+                                   SourceEditConsumer &EditConsumer,
+                                   SourceManager &SM) {
+    
+  if (funcDecl == nullptr) return;
+  
+  EditConsumer.accept(SM, startLocation, "/// <#Function Summary#>\n");
+  
+  processParameters(funcDecl->getParameterLists(), startLocation,
+                    EditConsumer, SM);
+  
+  if (funcDecl->hasThrows()) {
+    processThrows(startLocation, EditConsumer, SM);
+  }
+  
+  Type returnType = funcDecl->getReturnTypeLoc().getType();
+  processReturn(returnType, startLocation, EditConsumer, SM);
+  
+  EditConsumer.accept(SM, startLocation,"///\n");
+}
+    
+void generateDocCommentForConstructor(ConstructorDecl *constructorDecl,
+                                      swift::SourceLoc startLocation,
+                                      SourceEditConsumer &EditConsumer,
+                                      SourceManager &SM) {
+    
+  if (constructorDecl == nullptr) return;
+  
+  EditConsumer.accept(SM, startLocation, "/// <#Initializer Summary#>\n");
+  
+  //This extra step is needed for initializers to ensure the param
+  //self is not added to the documentation
+  ParameterList * allParams = constructorDecl->getParameters();
+  ArrayRef<const ParameterList *> allParamsList(allParams);
+  
+  processParameters(allParamsList, startLocation, EditConsumer, SM);
+  
+  if (constructorDecl->hasThrows()) {
+    processThrows(startLocation, EditConsumer, SM);
+  }
+  
+  EditConsumer.accept(SM, startLocation,"///\n");
+}
+    
+void generateDocCommentForSubscript(SubscriptDecl *subscriptDecl,
+                                    swift::SourceLoc startLocation,
+                                    SourceEditConsumer &EditConsumer,
+                                    SourceManager &SM) {
+  
+  if (subscriptDecl == nullptr) return;
+  
+  EditConsumer.accept(SM, startLocation, "/// <#Subscript Summary#>\n");
+  
+  processParameters(subscriptDecl->getIndices(), startLocation,
+                    EditConsumer, SM);
+  
+  Type returnType = subscriptDecl->getElementTypeLoc().getType();
+  processReturn(returnType, startLocation, EditConsumer, SM);
+  
+  EditConsumer.accept(SM, startLocation,"///\n");
+}
+   
+bool canGenerateDocComment(DeclKind kind) {
+  return kind == DeclKind::Func ||
+  kind == DeclKind::Constructor ||
+  kind == DeclKind::Subscript;
+}
+    
+bool RefactoringActionDocCommentBoilerplate::
+  isApplicable(ResolvedCursorInfo Tok, DiagnosticEngine &Diag) {
+  if (Tok.Kind != CursorInfoKind::ValueRef)
+    return false;
+  
+  DeclKind tokenKind = Tok.ValueD->getKind();
+  return canGenerateDocComment(tokenKind);
+}
+    
+bool RefactoringActionDocCommentBoilerplate::performChange() {
+  DeclKind tokenKind = CursorInfo.ValueD->getKind();
+  if (canGenerateDocComment(tokenKind)) return true;
```

Comment:
This line looks suspicious. We don't perform change if we can generate doc-comment?